File: 1207. Unique Number of Occurrences.py
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        s = set()

        mp = dict()

        for i in arr:
            mp[i] = mp.get(i, 0) + 1
        
        for key, val in mp.items():
            if val in s:
                return False
            s.add(val)
        return True

File: 1207. Unique Number of Occurrences.java
class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        HashMap<Integer, Integer> mp = new HashMap<>();
        HashSet<Integer> s = new HashSet<>();

        // Count occurrences of each number
        for (int i : arr) {
            mp.put(i, mp.getOrDefault(i, 0) + 1);
        }

        // Check if occurrences are unique
        for (int freq : mp.values()) {
            if (!s.add(freq)) { // If frequency is already in the set, return false
                return false;
            }
        }
        
        return true;
    }
}
